taxi/views.py

```diff
@@ -61,10 +63,24 @@ class CarListView(LoginRequiredMixin, generic.ListView):
 class CarDetailView(LoginRequiredMixin, generic.DetailView):
     model = Car
 
+    def post(self, request, *args, **kwargs):
+        car = self.get_object()
+        user = request.user
+
+        if "assign" in request.POST:
+            car.drivers.add(user)
+        elif "delete" in request.POST:
+            car.drivers.remove(user)
```

Comment:
In the `post` method of `CarDetailView`, consider checking if the user is already in the `drivers` before adding or removing them. This can prevent unnecessary database operations if the user is already assigned or not assigned to the car.